temporal-serviceclient/src/main/java/io/temporal/serviceclient/RpcRetryOptions.java

```diff
@@ -364,51 +344,63 @@ public RpcRetryOptions buildWithDefaultsFrom(RpcRetryOptions rpcRetryOptions) {
 
     /** Validates property values and builds RetryOptions with default values. */
     public RpcRetryOptions validateBuildWithDefaults() {
-      double backoff = backoffCoefficient;
-      if (backoff == 0d) {
-        backoff = DefaultStubServiceOperationRpcRetryOptions.BACKOFF;
+      double backoffCoefficient = this.backoffCoefficient;
+      if (backoffCoefficient < 1) {
+        backoffCoefficient = BACKOFF;
       }
-      if (initialInterval == null || initialInterval.isZero() || initialInterval.isNegative()) {
-        initialInterval = DefaultStubServiceOperationRpcRetryOptions.INITIAL_INTERVAL;
+      Duration initialInterval = this.initialInterval;
+      if (initialInterval == null) {
+        initialInterval = INITIAL_INTERVAL;
       }
-      if (congestionInitialInterval == null
-          || congestionInitialInterval.isZero()
-          || congestionInitialInterval.isNegative()) {
-        congestionInitialInterval =
-            DefaultStubServiceOperationRpcRetryOptions.CONGESTION_INITIAL_INTERVAL;
+      Duration congestionInitialInterval = this.congestionInitialInterval;
+      if (congestionInitialInterval == null) {
+        congestionInitialInterval = CONGESTION_INITIAL_INTERVAL;
       }
-      if (expiration == null || expiration.isZero() || expiration.isNegative()) {
-        expiration = DefaultStubServiceOperationRpcRetryOptions.EXPIRATION_INTERVAL;
+      Duration expiration = this.expiration;
+      if (expiration == null) {
+        expiration = EXPIRATION_INTERVAL;
       }
-
-      Duration maxInterval = this.maximumInterval;
-
-      if (maxInterval == null || maxInterval.isZero() || maxInterval.isNegative()) {
-        if (maximumAttempts == 0) {
-          maxInterval = DefaultStubServiceOperationRpcRetryOptions.MAXIMUM_INTERVAL;
-        } else {
-          maxInterval = null;
-        }
+      Duration maximumInterval = this.maximumInterval;
+      if (maximumInterval == null && maximumAttempts == 0) {
+        maximumInterval = initialInterval.multipliedBy(50);
```

Comment:
nit: Could we bring this constant `50` out as a `static final` so it's easier to identify what the default `maximumInterval` is?